### PR TITLE
5/Achievement Popup

### PR DESCRIPTION
**Pull request by jaden-z-fng**
Added new AchievementPopup class, which appears when an achievement is completed. 
(Based off of the work of @theboah , but he has yet to merge so the commit history is kinda ruined)

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/quests/AchievementPopup.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * AchievementPopup is an UIComponent that displays whenever the player completes a new achievement and then disappears.
+ */
+public class AchievementPopup extends UIComponent {
+    // Flag to see if popup is displaying.
+    private boolean showing;
+    // Image for achievement completion. */
+    private Image popup;
+
+
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/quests/AchievementPopup.java:25]: random space but fine

File: source/core/src/main/com/csse3200/game/components/quests/QuestManager.java [hunk 2]
```diff
@@ -162,7 +163,9 @@ private void addQuests(Task[] tasks, Map<DialogueKey, String[]> guideQuestDialog
         QuestBasic finalQuest = new QuestBasic("Final Boss", "Complete quest 1 and 2 to summon the boss", finalQuestTasks, false, null, null, false, false, 0);
         GameState.quests.quests.add(finalQuest);
 
-
+        // test achievement for functionality, remove on @theboah's merge
+        QuestHidden questHidden = new QuestHidden("steps", "these boots have seen everything");
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/quests/QuestManager.java:167]: I don't know if this does anything atm?

**Review by MargoCode — APPROVED** [review]
popup looks good  :)

**Review by LucasRyall — COMMENTED** [review]
Code looks good, but work on fixing the quests not triggering.

**Review by LucasRyall — APPROVED** [review]

**Pull request merged**